app/src/sharedTest/java/org/oppia/android/app/parser/HtmlParserTest.kt

```diff
@@ -271,6 +271,89 @@ class HtmlParserTest {
     assertThat(htmlResult.toString()).endsWith(" ")
   }
 
+  @Test
+  fun testHtmlContent_changeDeviceToLtr_textViewDirectionIsSetToLtr() {
+    val htmlParser = htmlParserFactory.create(
+      resourceBucketName,
+      entityType = "",
+      entityId = "",
+      imageCenterAlign = true
+    )
+    val textView = activityRule.scenario.runWithActivity {
+      it.getWindow().getDecorView().setLayoutDirection(View.LAYOUT_DIRECTION_LTR)
+      val textView: TextView = it.findViewById(R.id.test_html_content_text_view)
+      htmlParser.parseOppiaHtml(
+        "<p>You should know the following before going on:<br></p>" +
+          "<ul><li>The counting numbers (1, 2, 3, 4, 5 ….)<br></li>" +
+          "<li>How to tell whether one counting number is bigger or " +
+          "smaller than another<br></li></ul>",
+        textView
+      )
+      return@runWithActivity textView
+    }
+    assertThat(textView.layoutDirection).isEqualTo(View.LAYOUT_DIRECTION_LTR)
+  }
+
+  @Test
+  fun testHtmlContent_changeDeviceToRtl_textViewDirectionIsSetToRtl() {
+    val htmlParser = htmlParserFactory.create(
+      resourceBucketName,
+      entityType = "",
+      entityId = "",
+      imageCenterAlign = true
+    )
+    val textView = activityRule.scenario.runWithActivity {
+      it.getWindow().getDecorView().setLayoutDirection(View.LAYOUT_DIRECTION_RTL)
+      val textView: TextView = it.findViewById(R.id.test_html_content_text_view)
+      htmlParser.parseOppiaHtml(
+        "<p>You should know the following before going on:<br></p>" +
+          "<ul><li>The counting numbers (1, 2, 3, 4, 5 ….)<br></li>" +
+          "<li>How to tell whether one counting number is bigger or " +
+          "smaller than another<br></li></ul>",
+        textView
+      )
+      return@runWithActivity textView
+    }
+    assertThat(textView.layoutDirection).isEqualTo(View.LAYOUT_DIRECTION_RTL)
+  }
+
+  @Test
+  fun testHtmlContent_changeDeviceToRtlAndThenToLtr_textViewDirectionIsSetToRtlThenLtr() {
+    val htmlParser = htmlParserFactory.create(
+      resourceBucketName,
+      entityType = "",
+      entityId = "",
+      imageCenterAlign = true
+    )
+    val textViewLtr = activityRule.scenario.runWithActivity {
+      it.getWindow().getDecorView().setLayoutDirection(View.LAYOUT_DIRECTION_RTL)
+      val textView: TextView = it.findViewById(R.id.test_html_content_text_view)
+      htmlParser.parseOppiaHtml(
+        "<p>You should know the following before going on:<br></p>" +
+          "<ul><li>The counting numbers (1, 2, 3, 4, 5 ….)<br></li>" +
+          "<li>How to tell whether one counting number is bigger or " +
+          "smaller than another<br></li></ul>",
+        textView
+      )
+      return@runWithActivity textView
+    }
+    assertThat(textViewLtr.layoutDirection).isEqualTo(View.LAYOUT_DIRECTION_RTL)
```

Comment:
This extra assertion probably isn't needed because we know from the earlier test that setting RTL in the way above has this outcome (plus, with a helper per my other comment this test will become much simpler).

Reply:
Removed this assertion.